### PR TITLE
Install kn tasks only on architectures with knative client

### PR DESCRIPTION
**Pull request by yselkowitz**
# Changes

Which, currently, does not include ARM.  If/when that changes, it is suggested to simply add "arm64" instead of making these unconditional again (as was done in commit 4c80f347fadd2c4667aabe3412fbc120a1f692ca).

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by yselkowitz** [comment]
tests have passed
/assign @concaf

**Comment by nikhil-thomas** [comment]
@yselkowitz i see that this patch does two things:
1. disable installation of all kn, kn-apply ClusterTasks
2. skip the creation of +Add-flow Pipelines (default Pipelines) which use knative serving deployment as the deployment type 

@pradeepitm12 what do you think about the change in the pipeline generation part.


@yselkowitz  
1. could you confirm '2' is also an expected change.
1. could you add a release not mentioning both these changes.

File: pkg/reconciler/openshift/tektonaddon/pipelinetemplates/pipelinetemplates.go [hunk 1]
```diff
@@ -73,7 +74,14 @@ func GeneratePipelineTemplates(templatePath string, manifest *mf.Manifest) error
 	workspacedTaskGenerators := []taskGenerator{
 		&pipeline{environment: "openshift", nameSuffix: "", generateDeployTask: openshiftDeployTask},
 		&pipeline{environment: "kubernetes", nameSuffix: "-deployment", generateDeployTask: kubernetesDeployTask},
-		&pipeline{environment: "knative", nameSuffix: "-knative", generateDeployTask: knativeDeployTask},
+	}
+
+	// install knative tasks only where knative is available
```
Comment (pradeepitm12) [inline pkg/reconciler/openshift/tektonaddon/pipelinetemplates/pipelinetemplates.go:79]: :+1:

**Comment by yselkowitz** [comment]
>     1. could you confirm '2' is also an expected change.

Specifically, only on architectures which do not have OpenShift Serverless, namely `arm64`.  This matches what was done for `ppc64le` and `s390x` when they were in a similar state, and was then dropped by https://github.com/tektoncd/operator/commit/4c80f347fadd2c4667aabe3412fbc120a1f692ca once Serverless was available thereon.  Whenever Serverless does become available on ARM, instead of removing these conditionals again, I suggest that `"arm64"` simply be added to the list of knative architectures.

>     2. could you add a release not mentioning both these changes.

There should be no effect for existing users (which are on X/P/Z), this is only a prerequisite of the general ARM enablement.

**Comment by nikhil-thomas** [comment]
> `390x` when they were in a similar state, and was then dropped by [4c80f34](https://github.com/tektoncd/operator/commit/4c80f347fadd2c4667aabe3412fbc120a1f692ca) once Serverless was available thereon. Whenever Serverless does become available on ARM, instead of removing these conditionals again, I suggest that `"arm64"` simply be added to the list of knative architectures

Thanks 👍 
/lgtm
/approve

**Pull request merged**